typescript/sdk/src/token/xerc20.ts

```diff
@@ -0,0 +1,197 @@
+import { ethers } from 'ethers';
+import { Logger } from 'pino';
+import { Log, getAbiItem, parseEventLogs, toEventSelector } from 'viem';
+
+import { IXERC20Lockbox__factory } from '@hyperlane-xyz/core';
+import { Address, rootLogger } from '@hyperlane-xyz/utils';
+
+import {
+  GetEventLogsResponse,
+  getContractDeploymentTransaction,
+  getLogsFromEtherscanLikeExplorerAPI,
+} from '../block-explorer/etherscan.js';
+import { ExplorerFamily } from '../metadata/chainMetadataTypes.js';
+import { MultiProvider } from '../providers/MultiProvider.js';
+import { ChainNameOrId } from '../types.js';
+
+import { XERC20TokenExtraLockboxLimits } from './types.js';
+
+const minimalXERC20VSABI = [
+  {
+    anonymous: false,
+    inputs: [
+      {
+        indexed: true,
+        internalType: 'address',
+        name: 'bridge',
+        type: 'address',
+      },
+      {
+        indexed: false,
+        internalType: 'uint112',
+        name: 'bufferCap',
+        type: 'uint112',
+      },
+      {
+        indexed: false,
+        internalType: 'uint128',
+        name: 'rateLimitPerSecond',
+        type: 'uint128',
+      },
+    ],
+    name: 'ConfigurationChanged',
+    type: 'event',
+  },
+] as const;
+
+export type GetExtraLockboxesOptions = {
+  chain: ChainNameOrId;
+  xERC20Address: Address;
+  multiProvider: MultiProvider;
+  logger?: Logger;
+};
+
+export async function getExtraLockBoxConfigs({
+  xERC20Address,
+  chain,
+  multiProvider,
+  logger = rootLogger,
+}: GetExtraLockboxesOptions): Promise<XERC20TokenExtraLockboxLimits[]> {
+  const { apiUrl, family, apiKey } = multiProvider.getExplorerApi(chain);
+
+  // Fallback to use the rpc if the user has not provided an API key and the explorer family is Etherscan
+  // because the endpoint requires it or the explorer family is not known
+  const isExplorerConfiguredCorrectly =
+    family === ExplorerFamily.Etherscan ? !!apiKey : true;
+  const canUseExplorerApi =
+    family !== ExplorerFamily.Other && isExplorerConfiguredCorrectly;
+
+  const provider = multiProvider.getProvider(chain);
+  let startBlock = 0;
+  if (canUseExplorerApi) {
+    const contractDeploymentTx = await getContractDeploymentTransaction(
+      { apiUrl, apiKey: apiKey },
+      { contractAddress: xERC20Address },
+    );
+
+    const deploymentTransactionReceipt = await provider.getTransactionReceipt(
+      contractDeploymentTx.txHash,
+    );
+
+    startBlock = deploymentTransactionReceipt.blockNumber;
```

Comment:
When using etherscan, imo we don't need to add this logic, since we already filter on the event topic no? Simplifies the logic

Reply:
Unfortunately, it seems that we can't remove it because depending on the explorer API, it will start complaining that the block range params are required
![image](https://github.com/user-attachments/assets/e321eeac-853a-4cbb-bc45-b8b34a82e77a)